src/components/Common/UI/Switch/Switch.tsx

```diff
@@ -0,0 +1,68 @@
+import { Switch as _Switch, type SwitchProps as AriaSwitchProps } from 'react-aria-components'
+import { useFieldIds } from '../hooks/useFieldIds'
+import {
+  HorizontalFieldLayout,
+  type SharedHorizontalFieldLayoutProps,
+} from '../HorizontalFieldLayout'
+import styles from './Switch.module.scss'
+
+export interface SwitchProps
+  extends SharedHorizontalFieldLayoutProps,
+    Omit<AriaSwitchProps, 'children'> {
+  isSelected?: boolean
```

Comment:
Nit: thoughts on using `checked` instead as a name here for consistency with Radio and Checkbox? That makes more sense from a native html perspective (since it's internally a checkbox) but not sure if it's actually a very intuitive name for engineers